tests/test_username_api.py

```diff
@@ -40,405 +46,83 @@ def get_expected_response(website, user, status):
+      testcase_func_name=custom_name_func)
+  def test_available_username(self, website, user):
+    expected = get_expected_response(website, user, 404)
+    actual = get_response(self.app, website, user)
 
     # this is special to facebook checking,
     # and 'profile' field is only useful for
     # frontend purposes. Skipping here.
-    del actual['profile']
-
-    message = None
-    if expected != actual:
-      message = 'The provided available username ({}) returned 200'.format(user)
-
-      user = generate_random_username(25)
-      expected = get_expected_response('facebook', user, 404)
-      actual = get_response(self.app, 'facebook', user)
-
+    if website == 'facebook':
       del actual['profile']
 
-      if expected != actual:
-        message += ' and the random username ({}) returned 200'.format(user)
-      else:
-        message += ' but {} is still available'.format(user)
-
-    if message is not None:
-      logging.getLogger().warn(message)
-
-    assert expected == actual
-
-  @pytest.mark.parametrize('user', data['openhub']['available_usernames'])
-  def test_available_openhub_username(self, user):
-    expected = get_expected_response('openhub', user, 404)
-    actual = get_response(self.app, 'openhub', user)
-
     message = None
     if expected != actual:
-      message = 'The provided available username ({}) returned 200'.format(user)
-
-      user = generate_random_username(30, first_char_letter=True)
-      expected = get_expected_response('openhub', user, 404)
-      actual = get_response(self.app, 'openhub', user)
-
-      if expected != actual:
-        message += ' and the random username ({}) returned 200'.format(user)
-      else:
-        message += ' but {} is still available'.format(user)
-
-    if message is not None:
-      logging.getLogger().warn(message)
-
-    assert expected == actual
-
-  @pytest.mark.parametrize('user', data['deviantart']['available_usernames'])
-  def test_available_deviantart_username(self, user):
-    expected = get_expected_response('deviantart', user, 404)
-    actual = get_response(self.app, 'deviantart', user)
+      username_pattern = self.websites['username_patterns'][website]
+      username_length = 6
 
-    message = None
-    if expected != actual:
-      message = 'The provided available username ({}) returned 200'.format(user)
+      user = generate_random_username(website, username_length)
+      expected = get_expected_response(website, user, 404)
+      actual = get_response(self.app, website, user)
 
-      user = generate_random_username(15)
-      expected = get_expected_response('deviantart', user, 404)
-      actual = get_response(self.app, 'deviantart', user)
+      if website == 'facebook':
+        del actual['profile']
 
       if expected != actual:
-        message += ' and the random username ({}) returned 200'.format(user)
+        message = 'the random username ({}) returned 200'.format(user)
```

Comment:
why are you changing these messages?

Reply:
refer to @nalinbhardwaj's review
> Is there any point in having these messages now? We can already see the used username with parameterized, and know that it's failing...

Reply:
The function name cant be seen except in verbose mode.

Reply:
I thought function name will be shown if test fails. isn't it?

Reply:
It is shown, but not in the summary view, and you would be surprised how many 'developers' are scared of the pytest backtrace area (which can also be disabled; again, this concept of self-contained failure messages is relevant).